### PR TITLE
OPS-2495 agreement history changes

### PR DESCRIPTION
**Pull request by stevtek**
### BackEnd
* add transient property, `acting_change_request_id`, that associates a change to a BLI (or other entities in the future) with the change request behind the changes
* add `changes_requested_by_user_full_name` to history log items of Change Requests

### FrontEnd
* change the titles and messages of Agreement history log items as shown in the design for #2495.   These changes the log items of Change Requests.  Previously that looked just like edits with a statement added to the end saying that they were In Review.
* hide the Agreement History log items for changes to BLIs that had the changes also logged in a Change Request Approve/Decline log.
* fix: handle the lookup for services_component_id the Agreement history log.  These were defaulting just show IDs.

## Issue

#2495 

## How to test

* log in as COR User and make changes to status and budget that require approvals
* log in as Division Director and approve the changes
* also edit a Services Component on a BLI 
* view the Agreement History
 

### Status Changes
image

### Budget Changes
image

### Edits to Services Component
image

## Definition of Done Checklist
- [x] OESA: Code refactored for clarity
- [x] OESA: Dependency rules followed
- [x] Automated unit tests updated and passed
- [x] Automated integration tests updated and passed
- [x] Automated quality tests updated and passed
- [x] Automated load tests updated and passed
- [x] Automated a11y tests updated and passed
- [x] Automated security tests updated and passed
- [x] 90%+ Code coverage achieved
- ~[ ] Form validations updated~

## Links

_If relevant, e.g. for a link to a piece of markdown documentation_

### CODE REVIEW

File: frontend/src/helpers/utils.js [hunk 1]
```diff
@@ -151,7 +151,8 @@ export const codesToDisplayText = {
         date_needed: "Need By Date",
         line_description: "Description",
         proc_shop_fee_percentage: "Shop Fee",
-        status: "Status"
+        status: "Status",
+        services_component: "Services Component",
```
Comment (fpigeonjr) [inline frontend/src/helpers/utils.js:155]: 🤔 do we have a test we can update with this case?

File: frontend/src/hooks/lookup.hooks.js [hunk 2]
```diff
@@ -91,6 +91,24 @@ export const useGetNameForCanId = (id) => {
     return displayName;
 };
 
+/**
+ * This hook returns the display name of a Services Component given the id.
+ * @param {number} id - The id of the Services Component.
+ * @returns {string} - The display name of the Services Component.
+ */
+export const useGetNameForServicesComponentId = (id) => {
```
Comment (fpigeonjr) [inline frontend/src/hooks/lookup.hooks.js:99]: 🪝we already have this hook in this module: `src/hooks/useServicesComponents.hooks.js`

File: frontend/src/components/Agreements/AgreementDetails/AgreementHistoryList.jsx [hunk 3]
```diff
@@ -87,12 +88,13 @@ const propertyLogItemTitle = (logItem) => {
...
         title = `Budget Line ${propertyLabel} Edited`;
     } else if (logItem.event_class_name === "BudgetLineItemChangeRequest") {
+        title = logItem.property_key === "status" ? `Status Change to ${renderField(null, "status", logItem.change.new)}` : `Budget Change to ${propertyLabel}`;
         if (logItem.event_type === "IN_REVIEW") {
-            title = `Edit to BL ${propertyLabel} In Review`;
+            title += " In Review";
         } else if (logItem.event_type === "APPROVED") {
-            title = `Edit to BL ${propertyLabel} Approved`;
+            title += "  Approved";
         } else if (logItem.event_type === "REJECTED") {
```
Comment (fpigeonjr) [inline frontend/src/components/Agreements/AgreementDetails/AgreementHistoryList.jsx:96]: is it ok to have a mismatch here?

backend:REJECTED 
frontend: Declined

File: frontend/src/components/Agreements/AgreementDetails/AgreementHistoryList.jsx [hunk 4]
```diff
@@ -87,12 +88,13 @@ const propertyLogItemTitle = (logItem) => {
     if (logItem.event_class_name === "BudgetLineItem") {
         title = `Budget Line ${propertyLabel} Edited`;
     } else if (logItem.event_class_name === "BudgetLineItemChangeRequest") {
+        title = logItem.property_key === "status" ? `Status Change to ${renderField(null, "status", logItem.change.new)}` : `Budget Change to ${propertyLabel}`;
         if (logItem.event_type === "IN_REVIEW") {
```
Comment (fpigeonjr) [inline frontend/src/components/Agreements/AgreementDetails/AgreementHistoryList.jsx:92]: IN_REVIEW, APPROVED, and REJECTED/DECLINED may be a good use case for an ENUM

File: frontend/src/hooks/lookup.hooks.js (hunk 2 above)
Comment (stevtek) [reply frontend/src/hooks/lookup.hooks.js:99]: k, I'll get rid of mine and use that one

File: frontend/src/helpers/utils.js (hunk 1 above)
Comment (stevtek) [reply frontend/src/helpers/utils.js:155]: I'll can add a something to a unit test for this.
But if you have any ideas for e2e tests, let me know.  I wasn't if there was good place yet for that.

File: frontend/src/hooks/lookup.hooks.js (hunk 2 above)
Comment (stevtek) [reply frontend/src/hooks/lookup.hooks.js]: done

File: frontend/src/components/Agreements/AgreementDetails/AgreementHistoryList.jsx (hunk 4 above)
Comment (fpigeonjr) [reply frontend/src/components/Agreements/AgreementDetails/AgreementHistoryList.jsx:92]: example of a JS fake ENUM:
```js
const BLI_STATUS = {
    DRAFT: "DRAFT",
    PLANNED: "PLANNED",
    EXECUTING: "IN_EXECUTION",
    OBLIGATED: "OBLIGATED"
};
```

**Review by fpigeonjr — APPROVED** [review]
🤩 looked it over with @stevtek today and looks great!

File: backend/models/base.py [hunk 5]
```diff
@@ -131,6 +131,10 @@ def to_dict(self):
         schema = self.__marshmallow__()
         data = schema.dump(self)
         data["display_name"] = self.display_name
+        # add the transient attribute that tracks the change request responsible for changes (if it exists)
+        # so that it's added to the history event details
+        if hasattr(self, "acting_change_request_id"):
```
Comment (johndeange) [inline backend/models/base.py:136]: This would be better on a SQLAlchemy model specific `to_dict()` rather than on the `BaseModel`

**Review by johndeange — APPROVED** [review]

File: frontend/src/helpers/utils.js (hunk 1 above)
Comment (stevtek) [reply frontend/src/helpers/utils.js:155]: We've added e2e tests now, too.

File: backend/models/base.py (hunk 5 above)
Comment (stevtek) [reply backend/models/base.py:136]: I was thinking change requests could be used on any entity in the future.  For now it's just BLIs and next is Agreement if we do procurement shop changes.  But I can move to the BLI for now.

Comment (stevtek) [reply backend/models/base.py]: done

**Review by rajohnson90 — APPROVED** [review]

File: frontend/src/components/Agreements/AgreementDetails/AgreementHistoryList.jsx (hunk 4 above)
Comment (stevtek) [reply frontend/src/components/Agreements/AgreementDetails/AgreementHistoryList.jsx]: done

**Pull request merged**